### PR TITLE
Lift to upstream ResearchKit 2.2.15

### PR DESCRIPTION
**Pull request by philippzagar**
# Lift to upstream ResearchKit 2.2.15

## :recycle: Current situation & Problem
The current used ResearchKit version of our fork is 2.2.12, which is not up to date with the published ResearchKit in 2.2.15.
Furthermore, as described in #13, we currently don't validate within the CI if ResearchKit compiles with active Swift / C++ interop.

## :gear: Release Notes 
- Lift to ResearchKit 2.2.15
- Test within CI if ResearchKit compiles with active Swift / C++ interop

## :books: Documentation
Documented CI approach

## :white_check_mark: Testing
Built code (in C++ interop mode) via CI pipeline

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,11 +20,19 @@ on:
 
 jobs:
   build-xcarchive:
-    uses: StanfordBDHG/.github/.github/workflows/archive.yml@v2
+    uses: StanfordBDHG/.github/.github/workflows/archive.yml@feat/enable-interop-archiving
```
Comment (philippzagar) [inline .github/workflows/build.yml]: Note to ourselves: Adjust the checked out version as soon as https://github.com/StanfordBDHG/.github/pull/49 is merged

**Comment by philippzagar** [comment]
@PSchmiedmayer Please also adjust the merge checks of the repo

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for updating this to the latest version of the upstream ResearchKit repo! 👍

**Comment by PSchmiedmayer** [comment]
@philippzagar I adjusted the merge checks.

One thing that I was thinking was maybe adding a GitHub Note (in the formatting we have in some READMEs of our other repos) that notes how this repo is different from the main RK repo? I could see this right before the badges.

Maybe something like:
> NOTE:  
This repository is a fork of the [ResearchKit project by Apple](https://github.com/ResearchKit/ResearchKit), adding support for:
- The Swift Package Manager & the usage as a binary dependency speeding up build times ...
- Successful building using Swift's C++ Interoperability ...

Feel free to add and adapt this as you want to; serves more as a placeholder and idea.

**Comment by philippzagar** [comment]
> @philippzagar I adjusted the merge checks.
> 
> One thing that I was thinking was maybe adding a GitHub Note (in the formatting we have in some READMEs of our other repos) that notes how this repo is different from the main RK repo? I could see this right before the badges.
> 
> Maybe something like:
> 
> > NOTE:
> > This repository is a fork of the [ResearchKit project by Apple](https://github.com/ResearchKit/ResearchKit), adding support for:
> 
> * The Swift Package Manager & the usage as a binary dependency speeding up build times ...
> * Successful building using Swift's C++ Interoperability ...
> 
> Feel free to add and adapt this as you want to; serves more as a placeholder and idea.

Great idea @PSchmiedmayer, I think we have enough diff to upstream that this definitely makes sense!
I looked through the diff we currently have to upstream ResearchKit and the main two points (SPM support via binary target as well as C++ interop) were already stated by you. All the rest is not really noteworthy. Feel free to check out the [README](https://github.com/StanfordBDHG/ResearchKit/blob/feat/lift-to-rk-2.2.15/README.md)!

**Comment by philippzagar** [comment]
@PSchmiedmayer You also need to merge the PR as not all commits from upstream were signed.

**Comment by PSchmiedmayer** [comment]
Thank you @philippzagar; I will merge the PR 👍

**Pull request merged**